docs/080-breaking-changes.rst

```diff
@@ -49,6 +49,27 @@ New Restrictions
   3. Explicit conversions between literals and enums are only allowed if the literal can
      represent a value in the enum.
 
+* There are new restrictions on explicit type conversions. The conversion is only allowed when there
+  is at most one change in sign, width or type-category (``int``, ``address``, ``bytesNN``, etc.)
+  For example, the conversion ``uint16(int8)`` is disallowed since the conversion changes width (8
+  bits to 16 bits) and sign (signed integer to unsigned integer). To get the previous behaviour, use
+  an intermediate conversion. In the previous example, this would be ``uint16(uint8(int8))`` or
+  ``uint16(int16(int8))``. Note that the two ways to convert will produce different results e.g.,
+  for ``-1``. The following are some examples of conversions that are disallowed by this rule. Note
+  that, given types ``T`` and ``S``, the notation ``T(S)`` refers to the explicit conversion
```

Comment:
We can also use this notation above if you move the explanation up.

Reply:
Then, I'll just move the explanation up.